PharmacyManagement/Commodity/AllCommodities.cs

```diff
@@ -1,21 +1,155 @@
+            dataTable.Fill(dgvCmd);
+            BindingSource binding = new BindingSource();
+            binding.DataSource = dataTable;
+
+            dgvAllCommodities.DataSource = binding;
+            bindingNavigator.BindingSource = binding;
+
+            txtCommodityID.DataBindings.Clear();
+            cboCommodityType.DataBindings.Clear();
+            txtCommodityName.DataBindings.Clear();
+            txtManufacturer.DataBindings.Clear();
+            txtQuantity.DataBindings.Clear();
+            txtBaseUnit.DataBindings.Clear();
+            txtPurchasePrice.DataBindings.Clear();
+            txtSellingPrice.DataBindings.Clear();
+            dtpMfgDate.DataBindings.Clear();
+            dtpExpDate.DataBindings.Clear();
+
+            txtCommodityID.DataBindings.Add("Text", binding, "CommodityID");
+            cboCommodityType.DataBindings.Add("SelectedValue", binding, "CategoryID");
+            txtCommodityName.DataBindings.Add("Text", binding, "CommodityName");
+            txtManufacturer.DataBindings.Add("Text", binding, "Manufacturer");
+            txtQuantity.DataBindings.Add("Text", binding, "Quantity");
+            txtBaseUnit.DataBindings.Add("Text", binding, "BaseUnit");
+            txtPurchasePrice.DataBindings.Add("Text", binding, "PurchasePrice");
+            txtSellingPrice.DataBindings.Add("Text", binding, "SellingPrice");
+            dtpMfgDate.DataBindings.Add("Value", binding, "MfgDate");
+            dtpExpDate.DataBindings.Add("Value", binding, "ExpDate");
+        }
+
+        private void ToggleControls(bool value)
+        {
+            
+            txtCommodityID.Enabled = false;
+            btnReload.Enabled = true;
+
+            cboCommodityType.Enabled = value;
+            txtCommodityName.Enabled = value;
+            txtManufacturer.Enabled = value;
+            txtQuantity.Enabled = value;
+            txtBaseUnit.Enabled = value;
+            txtPurchasePrice.Enabled = value;
+            txtSellingPrice.Enabled = value;
+            dtpMfgDate.Enabled = value;
+            dtpExpDate.Enabled = value;
+
+            btnSave.Enabled = value;
+
+            btnEdit.Enabled = !value;
+            btnDelete.Enabled = !value;
+        }
+
+        private void btnSave_Click(object sender, EventArgs e)
+        {
+            UpdateCommodity();
+            MessageBox.Show("Updated information successfully", "Update", MessageBoxButtons.OK, MessageBoxIcon.Information);
+            AllCommodities_Load(sender, e);
+        }
+
+        private void UpdateCommodity()
+        {
+            string updateCommodityQuery = @"UPDATE COMMODITY
+                                   SET CommodityName = @newCommodityName,
+                                       Manufacturer = @Manufacturer,
+                                       Quantity = @Quantity,
+                                       BaseUnit = @BaseUnit,
+                                       PurchasePrice = @PurchasePrice,
```

Comment:
format price N0 and base unit of price is VND

Reply:
I replace it at SELECT script